modules/custom/openy_upgrade_tool/src/OpenyUpgradeLogManager.php

```diff
@@ -0,0 +1,588 @@
+<?php
+
+namespace Drupal\openy_upgrade_tool;
+
+use Drupal\config\Form\ConfigSync;
+use Drupal\config\StorageReplaceDataWrapper;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ConfigImporter;
+use Drupal\Core\Config\ConfigImporterException;
+use Drupal\Core\Config\ConfigManagerInterface;
+use Drupal\Core\Config\StorageComparer;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Config\TypedConfigManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Extension\ModuleInstallerInterface;
+use Drupal\Core\Extension\ThemeHandlerInterface;
+use Drupal\Core\Lock\LockBackendInterface;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\Core\Url;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\features\FeaturesManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Wrapper for main logic that related to OpenyUpgradeLog.
+ */
+class OpenyUpgradeLogManager implements OpenyUpgradeLogManagerInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Upgrade path dashboard route.
+   */
+  const DASHBOARD = 'entity.openy_upgrade_log.collection';
+
+  /**
+   * Upgrade path dashboard route.
+   */
+  const MODAL_WIDTH = 1000;
+
+  /**
+   * The FeaturesManager.
+   *
+   * @var \Drupal\features\FeaturesManagerInterface
+   */
+  public $featuresManager;
+
+  /**
+   * Entity type manger.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * Logger Entity Storage.
+   *
+   * @var \Drupal\Core\Entity\Sql\SqlContentEntityStorage
+   */
+  public $loggerEntityStorage = NULL;
```

Comment:
what sense makes default of NULL for the storage?
We are not removing logger entity
Please fix architecture to meet requirement

Reply:
Initially `loggerEntityStorage` setup in `__construct` but this fail build on site install build because of `getLoggerEntityTypeName` method, so this was moved to `getLoggerEntityStorage()` method:

```php
  /**
   * Get entity storage that used for upgrade tool.
   */
  public function getLoggerEntityStorage() {
    // TODO: Delete this, after deprecated 'logger_entity' deleting
    // and create instance in __construct.
    if ($this->loggerEntityStorage) {
      return $this->loggerEntityStorage;
    }

    $this->loggerEntityStorage = $this->entityTypeManager
      ->getStorage($this->getLoggerEntityTypeName());
    return $this->loggerEntityStorage;
  }
```